homeassistant/components/overkiz/config_flow.py

```diff
@@ -46,7 +46,7 @@ async def async_validate_input(self, user_input: dict[str, Any]) -> None:
         username = user_input[CONF_USERNAME]
         password = user_input[CONF_PASSWORD]
         server = SUPPORTED_SERVERS[user_input[CONF_HUB]]
-        session = async_get_clientsession(self.hass)
+        session = async_create_clientsession(self.hass)
```

Comment:
This one will need to be detached manually since auto cleanup won't work here since its not being called from a specific config entry

```
    """Create a new ClientSession with kwargs, i.e. for cookies.

    If auto_cleanup is False, you need to call detach() after the session
    returned is no longer used. Default is True, the session will be
    automatically detached on homeassistant_stop or when being created
    in config entry setup, the config entry is unloaded.

    This method must be run in the event loop.
    """
```

Reply:
Good catch, thanks!

Reply:
Fixed in 36e3593f6b7273c8356d423c9cd7b57954a20822

See https://github.com/iMicknl/python-overkiz-api/blob/5c84b142dc77255d7a4d01433bbcf609f7318d55/pyoverkiz/client.py#L128-L135 where we automatically close the session. Or can we only detach() it?

Reply:
What you are doing is fine there. Home Assistant overrides the close method so the right thing will happen, and if you use it outside of Home Assistant, the right thing though happen there as well.